src/test/java/com/teragrep/cfe_39/Ingestion0FilesLowSizeTest.java

```diff
@@ -0,0 +1,190 @@
+import com.teragrep.cfe_39.consumers.kafka.HdfsDataIngestion;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Ingestion0FilesLowSizeTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Ingestion0FilesTest.class);
+    private static MiniDFSCluster hdfsCluster;
+    private static File baseDir;
+    private static Config config;
+    private FileSystem fs;
+
+    // Prepares known state for testing.
+    @BeforeEach
+    public void startMiniCluster() {
+        assertDoesNotThrow(() -> {
+            // Set system properties to use the valid configuration.
+            System
+                    .setProperty("cfe_39.config.location", System.getProperty("user.dir") + "/src/test/resources/valid.application.properties");
+            config = new Config();
+            // Create a HDFS miniCluster
+            baseDir = Files.createTempDirectory("test_hdfs").toFile().getAbsoluteFile();
+            hdfsCluster = new TestMiniClusterFactory().create(config, baseDir);
+            config = new Config("hdfs://localhost:" + hdfsCluster.getNameNodePort() + "/", 3000);
+            fs = new TestFileSystemFactory().create(config.getHdfsuri());
+        });
+    }
+
+    // Teardown the minicluster
+    @AfterEach
+    public void teardownMiniCluster() {
+        assertDoesNotThrow(() -> {
+            fs.close();
+        });
+        hdfsCluster.shutdown();
+        FileUtil.fullyDelete(baseDir);
+    }
+
+    @DisabledIfSystemProperty(
+            named = "skipIngestionTest",
+            matches = "true"
+    )
+    @Test
+    public void ingestion0FilesLowSizeTest() {
+        /*This test case is for testing the functionality of the ingestion when there are files already present in the database before starting ingestion.
+        Maximum file size is set to 3,000 in the config.
+        Empty HDFS database, 140 records in mock kafka consumer ready for ingestion. All 14 records for each 10 topic partitions are stored in two avro-files per partition based on MaximumFileSize.*/
+        assertDoesNotThrow(() -> {
+            Assertions.assertTrue(config.getPruneOffset() >= 300000L); // Fails the test if the config is not correct.
+            Assertions.assertFalse(fs.exists(new Path(config.getHdfsPath() + "/" + "testConsumerTopic")));
+            HdfsDataIngestion hdfsDataIngestion = new HdfsDataIngestion(config);
+            Thread.sleep(10000);
```

Comment:
Add a comment explaining the sleep call.

Reply:
Solved in commit a928f4d3941114b85f97cfa0bd08a886de54aab6